prow/cmd/peribolos/main.go

```diff
@@ -255,19 +255,78 @@ func configureOrg(opt options, client *github.Client, orgName string, orgConfig
 
 	// teams = set(all teams)
 	for name, team := range orgConfig.Teams {
-		team.Name = name
-		num := -1
-		// find team name in teams (or previous name)
-		if err := configureTeam(client, num, team); err != nil {
+		_ = name
+		if err := configureTeam(client, orgName, team); err != nil {
 			return fmt.Errorf("failed to update %s: %v", orgName, err)
 		}
 	}
 	return nil
 }
 
-func configureTeam(client *github.Client, num int, team org.Team) error {
-	// create team if num < 0
-	// else diff and patch
+func findTeam(client *github.Client, org, name string, previousNames ...string) (*github.Team, error) {
+	teams, err := client.ListTeams(org)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list %s teams: %v", org, err)
+	}
+	// Does a team exactly match the current name?
+	for _, t := range teams {
+		if t.Name == name { // Yep!
+			return &t, nil
+		}
+	}
+	// Maybe one of the previous names is around?
+	names := sets.NewString(previousNames...)
+	for _, t := range teams {
+		if names.Has(t.Name) { // Yep!
+			return &t, nil
+		}
+	}
+
+	// Team does not exist
+	return nil, nil
+}
+
+// shouldPatch will return true if any non-nil values in org.Team differ from the github.Team values.
+func shouldPatch(gt github.Team, ot org.Team) bool {
+	if gt.Name != ot.Name {
+		return true
+	}
+	if ot.Description != nil && gt.Description != *ot.Description {
+		return true
+	}
+	if ot.Privacy != nil && gt.Privacy != string(*ot.Privacy) {
+		return true
+	}
+	return false
+}
+
+// configureTeam creates/updates the will find the team id that best matches the name
+func configureTeam(client *github.Client, orgName string, team org.Team) error {
+	// What team is the best match for the name?
+	gt, err := findTeam(client, orgName, team.Name, team.Previously...)
+	if err != nil {
+		return err
+	}
+	if gt == nil { // Team doesn't exist, create it
+		gt, err = client.CreateTeam(orgName, team.Name)
```

Comment:
I see how this makes the edit logic nice in this method but why do two client calls here (create, edit) instead of one (create with all info)?

Reply:
Done